datalad/config.py

```diff
@@ -232,10 +232,16 @@ def __init__(self, dataset=None, overrides=None, source='any'):
         self._src_mode = source
         run_kwargs = dict()
         if dataset is not None:
-            # make sure we run the git config calls in the dataset
-            # to pick up the right config files
-            run_kwargs['cwd'] = dataset.path
-        self._runner = GitWitlessRunner(**run_kwargs)
+            if hasattr(dataset, '_git_runner'):
+                self._runner = dataset._git_runner
+            elif dataset.repo:
+                self._runner = dataset.repo._git_runner
+            else:
+                # make sure we run the git config calls in the dataset
+                # to pick up the right config files
+                run_kwargs['cwd'] = dataset.path
```

Comment:
I think I'm missing something because this code seems to be covered and didn't error, but doesn't this arm leave `_runner` undefined?

Reply:
I think you're right, @kyleam. Thanks for spotting that.
It might not error, because this only happens, when we instantiate a `Dataset.config` for a dataset that wasn't created yet (has no `.repo`), so we probably never actually call anything on it in the tests.

Reply:
This arm should do the same thing as the outer `else`.